### PR TITLE
chore(themed): add token CSS variable test helper

### PR DESCRIPTION
**Pull request by alisonailea**
**Related Issue:** #7180 

## Summary

Includes helpers which assist developers in testing and theming component tokens as CSS variables.

#### Demo page

```html
<demo-theme tokens="--calcite-button-background-color, --calcite-button-border-color, --calcite-button-text-color"
    ><calcite-button kind="inverse" scale="l"> Button </calcite-button
></demo-theme>
```

#### Storybook / Chromatic

```ts
export const theming_TestOnly = (): string =>
  html` <style>
      .container {
        ${setCSSVariables([
        "--calcite-button-background-color",
        "--calcite-button-border-color",
        "--calcite-button-text-color",
      ])}
      }
    </style>
    <div class="container">
        <calcite-button kind="inverse" scale="l"> Button </calcite-button>
   </div>
`;
```

#### E2E

```ts
describe("theme", () => {
  themed(`<calcite-button kind="inverse" scale="l"> Button </calcite-button>`,  {
    "--calcite-button-background-color": {
      shadowSelector: ".button",
      targetProp: "backgroundColor",
    },
    "--calcite-button-text-color": {
      shadowSelector: ".button",
      targetProp: "color",
    }
    "--calcite-accordion-border-color": {
      shadowSelector: ".button",
      targetProp: "borderColor",
    },
  });
});
```

### CODE REVIEW

**Review by jcfranco — APPROVED** [review]
👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾
👾🪙🪙🪙🪙👾🪙👾👾👾👾👾🪙👾🪙🪙🪙🪙👾👾🪙🪙🪙👾👾🪙🪙👾👾🪙👾👾👾🪙👾🪙🪙🪙🪙👾🪙👾
👾🪙👾👾🪙👾🪙👾👾👾👾👾🪙👾🪙👾👾👾👾🪙👾👾👾👾🪙👾👾🪙👾🪙🪙👾🪙🪙👾🪙👾👾👾👾🪙👾
👾🪙🪙🪙🪙👾🪙👾👾🪙👾👾🪙👾🪙🪙🪙👾👾👾🪙🪙👾👾🪙👾👾🪙👾🪙👾🪙👾🪙👾🪙🪙🪙👾👾🪙👾
👾🪙👾👾🪙👾🪙👾🪙👾🪙👾🪙👾🪙👾👾👾👾👾👾👾🪙👾🪙👾👾🪙👾🪙👾👾👾🪙👾🪙👾👾👾👾👾👾
👾🪙👾👾🪙👾👾🪙👾👾👾🪙👾👾🪙🪙🪙🪙👾🪙🪙🪙👾👾👾🪙🪙👾👾🪙👾👾👾🪙👾🪙🪙🪙🪙👾🪙👾
👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾👾

This should be good to merge once comments are addressed.

**Comment by jcfranco** [comment]
Before merging, can you update the PR title to "...: _add_ token CSS ...", so it follows [conventions](https://github.com/Esri/calcite-design-system/blob/main/CONTRIBUTING.md#descriptive-commit-subject)?

**Comment by alisonailea** [comment]
I've had to duplicate the theme code because I can't find a folder that the demos and storybook servers share.

**Pull request merged**
